dev-client/src/screens/SoilScreen/ColorScreen/components/ManualWorkflow.tsx

```diff
@@ -47,103 +46,59 @@ import {
 import Animated, {LinearTransition} from 'react-native-reanimated';
 import {StyleSheet} from 'react-native';
 import {
-  isValidChromaFor,
-  isValidSubstepFor,
-  isValidValueFor,
-  validComponents,
-} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/soilColors';
+  ColorUpdate,
+  PartialColor,
+  updateColorSelections,
+  validProperties,
+} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/soilColorValidation';
 
 export const ManualWorkflow = (props: SoilPitInputScreenProps) => {
   const {t} = useTranslation();
   const dispatch = useDispatch();
   const data = useSelector(selectDepthDependentData(props));
   const {complete} = pitMethodSummary(t, data, 'soilColor');
 
-  const {substep: initialSubstep, hue: initialHue} = useMemo(
-    () =>
-      typeof data?.colorHue === 'number'
-        ? renderMunsellHue({
-            colorHue: data?.colorHue,
-            colorChroma: data?.colorChroma,
-          })
-        : {substep: null, hue: null},
-    [data],
-  );
+  const {hue: initialHue, substep: initialSubstep} =
+    typeof data?.colorHue === 'number'
```

Comment:
comment: i feel that this business of inspecting the structure of the munsell data and deciding how to render it could be factored out to a helper method somewhere